snuba/datasets/querylog_processor.py

```diff
@@ -49,13 +55,27 @@ def __extract_query_list(
             cache_hit.append(int(query["stats"].get("cache_hit") or 0))
             sample.append(query["stats"].get("sample") or 0)
             max_threads.append(query["stats"].get("max_threads") or 0)
-            num_days.append(query["stats"].get("num_days") or 0)
             clickhouse_table.append(query["stats"].get("clickhouse_table") or "")
             query_id.append(query["stats"].get("query_id") or "")
             # XXX: ``is_duplicate`` is currently not set when using the
             # ``Cache.get_readthrough`` query execution path. See GH-902.
             is_duplicate.append(int(query["stats"].get("is_duplicate") or 0))
             consistent.append(int(query["stats"].get("consistent") or 0))
+            profile = query.get("profile") or {
+                "time_range": 0,
+                "all_columns": [],
+                "multi_level_condition": False,
+                "where_profile": {"columns": [], "mapping_cols": []},
+                "groupby_cols": [],
+                "array_join_cols": [],
+            }
```

Comment:
There's also a "table" attribute added on ClickhouseQueryProfile - is that intentionally omitted here?

In what scenarios would `query.get("profile")` not be present? It's not marked as optional here https://github.com/getsentry/snuba/blob/master/snuba/querylog/query_metadata.py#L70

`time_range` is also marked as optional in ClickhouseQueryProfile - should that be set to `0` if no value is provided

Reply:
> There's also a "table" attribute added on ClickhouseQueryProfile - is that intentionally omitted here?

I skipped when I realized there was already one in the table. Did not see that before.

> In what scenarios would query.get("profile") not be present? It's not marked as optional here https://github.com/getsentry/snuba/blob/master/snuba/querylog/query_metadata.py#L70

Just for the rollout. This PR is both going to write on kafka and consume. So there may be a moment in time where the consumer will be reading and trying to extract the profile but the producers will still be running the old code and not adding the profile. I could deploy them separately, but the problem could still happen since I would have to verify that the consumer is only consuming new messages before deploying the consumer.

> time_range is also marked as optional in ClickhouseQueryProfile - should that be set to 0 if no value is provided?

It is optional because I wanted to distinguish the case of queries shorter than one day from errors where we could not extract a valid time range from the query. Though I realized then that we already had the column for that and did not want to make it nullable. Those errors are just bugs in the profiler code (all queries have a time range) so writing 0 in those case is no big deal.